modules/ROOT/pages/_partials/commons/common-vector-search.adoc

```diff
@@ -0,0 +1,217 @@
+
+* Cost Per Query Reduction - When you have hundreds of thousands of connected clients querying against a cloud-based LLM, the load on cloud model and operational costs of running the cloud based model can be considerably high.
+By running queries locally on the device, you can save on data transfer costs and cloud egress charges while also decentralizing the operational costs.
+
+== Key Concepts of Vector Search in Couchbase Lite 
+
+When xref:{param-module}:working-with-vector-search.adoc[working with Vector Search], you should be aware of the core concepts below. 
+
+=== About Vector Embeddings
+
+Vector embeddings represent the output of a Machine Learning (ML) model as an array of numbers to capture semantic or contextual relationships between data points.
+This representation encodes how a ML model understands the input or inputs provided to it, based on how the model was initially trained and the internal structure of the model.
+When a model considers the features of a given input as similar, the distance between the vector embeddings will be short.
+Vector embeddings are stored within embedded vector indexes.
+
+The current supported formats for Vector embeddings are:
+
+* An array of 32-bit floats.
+
+* A Base64 string that encodes a https://en.wikipedia.org/wiki/Endianness[Little-endian] array of 32-bit floats.
+
+=== About Vector Indexes
+
+Vector Indexes are used to store and manage vector representations of content in the form of vector embeddings.
+You can use Vector Indexes to efficiently retrieve vectors, similar to a target vector.
+Before use, a Vector Index needs to be trained to compute the centroids and parameters for encoding the vectors.
+
+NOTE: If there are insufficient vectors in the database for training, a warning message will be logged, indicating the required number of vectors.
+
+Training is configured to commence automatically on the first Vector Search query when there are enough vectors to be trained based on the `minimum-training-size` configuration.
+Be aware that vector index training can affect query performance.
+If a query is executed against the index before it is trained, a full scan of the vectors will be performed. 
+
+
+
+=== About Lazy Vector Indexes
+
+IMPORTANT: Lazy index is not an automatic process, you will need to manually schedule the index updates. 
+
+Lazy vector indexes (lazy index) is Couchbase Lite specific functionality that updates indexes asynchronously, satisfying the following use cases:
+
+* Documents have been added to the application by the end-user with no available Machine Learning (ML) model to generate the vectors.
+You can use lazy indexing to schedule updates to the index of said documents when a ML model is available. 
+
+* The remote ML model used stops working or has intermittent availability, causing a failed update.
+With lazy index, you can skip documents that fail to update and schedule the index process to be retried at a later date.
```

Comment:
With lazy indexing, you can skip documents that fail to update and schedule the index process at a later date.

Reply:
Done!